resources/lib/streamservice.py

```diff
@@ -264,18 +264,23 @@ def _handle_stream_api_error(self, video_json, message):
         self._kodi.show_ok_dialog(message=message)
         self._kodi.end_of_directory()
 
-    def _handle_bad_stream_error(self, stream_type):
-        ''' Show a localized error message in Kodi GUI for a failing VRT NU stream based on stream type: HLS/MPEG-DASH)
-            message: VRT NU stream <stream_type> problem, try again with InputStream Adaptive enabled/disabled: 30959=disabled, 30960=enabled
+    def _handle_bad_stream_error(self, protocol):
+        ''' Show a localized error message in Kodi GUI for a failing VRT NU stream based on protocol: hls, hls_aes, mpeg_dash)
+            message: VRT NU stream <stream_type> problem, try again with (InputStream Adaptive) (and) (DRM) enabled/disabled: 30959=and DRM, 30960=disabled, 30961=enabled
         '''
-        if self._kodi.has_inputstream_adaptive():
-            message = self._kodi.localize(30958) % (stream_type, self._kodi.localize(30959))
+        # HLS AES DRM failed
+        if protocol == 'hls_aes' and not self._kodi.has_inputstream_adaptive() and self._kodi.get_setting('usedrm', 'true') == 'false':
```

Comment:
That is quite advanced, I wouldn't probably take it this far, but it's definitely much clearer.